src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -33,12 +44,219 @@ public static class DefaultAssemblyPartDiscoveryProvider
+            var filePath = Path.Combine(Path.GetDirectoryName(assembly.Location), fileName);
+            var additionalAssembly = LoadAssembly(filePath);
+
+            if (additionalAssembly == null)
+            {
+                return null;
+            }
+
+            return additionalAssembly;
+        }
+
+        private static (string metadataPath, string metadataIncludeByDefault) ParseMetadataAttribute(
+            AssemblyMetadataAttribute metadataAttribute)
+        {
+            var data = metadataAttribute.Value.Split(MetadataSeparators);
+            if (data.Length != 2 || string.IsNullOrWhiteSpace(data[0]) || string.IsNullOrWhiteSpace(data[1]))
+            {
+                return default;
+            }
+
+            return (data[0], data[1]);
+        }
+
+        private static Assembly LoadAssembly(string filePath)
+        {
+            Assembly viewsAssembly = null;
+            if (AssemblyResolver(filePath))
+            {
+                try
+                {
+                    viewsAssembly = AssemblyLoader(filePath);
+                }
+                catch (FileLoadException)
+                {
+                    // Don't throw if assembly cannot be loaded. This can happen if the file is not a managed assembly.
+                }
+            }
+
+            return viewsAssembly;
+        }
+
+        private static Assembly LoadFromConvention(Assembly assembly)
+        {
+            if (assembly.IsDynamic || string.IsNullOrEmpty(assembly.Location))
+            {
+                return null;
+            }
+
+            for (var i = 0; i < ViewAssemblySuffixes.Count; i++)
+            {
+                var fileName = assembly.GetName().Name + ViewAssemblySuffixes[i] + ".dll";
+                var filePath = Path.Combine(Path.GetDirectoryName(assembly.Location), fileName);
+
+                var viewsAssembly = LoadAssembly(filePath);
+                if (viewsAssembly != null)
+                {
+                    return viewsAssembly;
+                }
+            }
+
+            return null;
+        }
+
+        private static void AddAdditionalReference(
+            Assembly additionalReference,
+            Assembly entryAssembly,
+            Assembly assembly,
+            SortedSet<Assembly> additionalReferences,
+            SortedSet<Assembly> entryAssemblyAdditionalReferences)
+        {
+            if (additionalReference == null ||
+                additionalReferences.Contains(additionalReference) ||
+                entryAssemblyAdditionalReferences.Contains(additionalReference))
+            {
+                return;
+            }
+
+            if (assembly.Equals(entryAssembly))
+            {
+                entryAssemblyAdditionalReferences.Add(additionalReference);
+            }
+            else
+            {
+                additionalReferences.Add(additionalReference);
+            }
+        }
+
+        internal class AdditionalReferencesPair
```

Comment:
Is this just to avoid the test failures on net46?

Reply:
Yes